test/new-e2e/tests/otel/utils/pipelines_utils.go

```diff
@@ -319,6 +402,78 @@ func TestPrometheusMetrics(s OTelTestSuite) {
 	s.T().Log("Got otelcol_datadog_trace_agent_trace_writer_spans", traceAgentMetrics)
 }
 
+// TestSpanReceiverV2 tests that APM stats are correct when using probabilistic sampling
+func TestSpanReceiverV2(s OTelTestSuite) {
+	ctx := context.Background()
+	err := s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+	require.NoError(s.T(), err)
+	numTraces := 10
+
+	s.T().Log("Starting telemetrygen")
+	createTelemetrygenJobForSpanReceiverV2(ctx, s, "traces", []string{"--traces", fmt.Sprint(numTraces)})
+}
+
+func createTelemetrygenJobForSpanReceiverV2(ctx context.Context, s OTelTestSuite, telemetry string, options []string) {
+	var ttlSecondsAfterFinished int32 = 0 //nolint:revive // We want to see this is explicitly set to 0
+	var backOffLimit int32 = 4
+
+	otlpEndpoint := fmt.Sprintf("%v:4317", s.Env().Agent.LinuxNodeAgent.LabelSelectors["app"])
+	jobSpec := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("telemetrygen-job-%v-%v", telemetry, strings.ReplaceAll(strings.ToLower(s.T().Name()), "/", "-")),
+			Namespace: "datadog",
+		},
+		Spec: batchv1.JobSpec{
+			TTLSecondsAfterFinished: &ttlSecondsAfterFinished,
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Env: []corev1.EnvVar{{
+								Name:  "OTEL_SERVICE_NAME",
+								Value: telemetrygenService,
+							}, {
+								Name:      "OTEL_K8S_POD_ID",
+								ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.uid"}},
+							}, {
+								Name:      "OTEL_K8S_NAMESPACE",
+								ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.namespace"}},
+							}, {
+								Name:      "OTEL_K8S_NODE_NAME",
+								ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "spec.nodeName"}},
+							}, {
+								Name:      "OTEL_K8S_POD_NAME",
+								ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.name"}},
+							}},
+							Name:  "telemetrygen-job",
+							Image: "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:v0.107.0",
+							Command: append([]string{
+								"/telemetrygen", telemetry, "--otlp-endpoint", otlpEndpoint, "--otlp-insecure",
+								"--telemetry-attributes", fmt.Sprintf("%v=%v", customAttribute, customAttributeValue),
+								"--telemetry-attributes", fmt.Sprintf("%v=\"%v\"", "k8s.pod.uid", "did-it-work"),
```

Comment:
If this is the only change in this new telemetrygen function, could we modify `createTelemetrygenJob` instead of creating a new one? We can remove `k8s.pod.uid` from the default command list and pass it in as an option instead:
- `"--otlp-attributes", "k8s.pod.uid=\"$(OTEL_K8S_POD_ID)\""` for the existing sampling test
- `"--telemetry-attributes", "k8s.pod.uid=\"$(OTEL_K8S_POD_ID)\""` for your test

Also, I don't think we need to use `"did-it-work"` since we're just checking that container id is empty.